packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -0,0 +1,834 @@
+   * @param params.userId - user email or id from Social login
+   * @param params.password - The password used to create new wallet and seedphrase
+   * @param params.seedPhrase - The seed phrase to backup
+   * @returns A promise that resolves to the encrypted seed phrase and the encryption key.
+   */
+  async createToprfKeyAndBackupSeedPhrase({
+    authConnectionId,
+    groupedAuthConnectionId,
+    userId,
+    password,
+    seedPhrase,
+  }: {
+    authConnectionId: string;
+    groupedAuthConnectionId?: string;
+    userId: string;
+    password: string;
+    seedPhrase: Uint8Array;
+  }): Promise<void> {
+    // locally evaluate the encryption key from the password
+    const { encKey, authKeyPair, oprfKey } = this.toprfClient.createLocalKey({
+      password,
+    });
+
+    // encrypt and store the seed phrase backup
+    await this.#encryptAndStoreSeedPhraseBackup(
+      seedPhrase,
+      encKey,
+      authKeyPair,
+    );
+
+    // store/persist the encryption key shares
+    // We store the seed phrase metadata in the metadata store first. If this operation fails,
+    // we avoid persisting the encryption key shares to prevent a situation where a user appears
+    // to have an account but with no associated data.
+    await this.#persistOprfKey({
+      groupedAuthConnectionId,
+      authConnectionId,
+      userId,
+      oprfKey,
+      authPubKey: authKeyPair.pk,
+    });
+    // TODO: store srp hashes or some identifier for srp in array to see if a srp is part of backup
+    // create a new vault with the resulting authentication data
+    await this.#createNewVaultWithAuthData({
+      password,
+      rawToprfEncryptionKey: encKey,
+      rawToprfAuthKeyPair: authKeyPair,
+    });
+  }
+
+  /**
+   * Add a new seed phrase backup to the metadata store.
+   *
+   * @param seedPhrase - The seed phrase to backup.
+   * @param password - The password used to create new wallet and seedphrase
+   * @returns A promise that resolves to the success of the operation.
+   */
+  async addNewSeedPhraseBackup(
+    seedPhrase: Uint8Array,
+    password: string, // TODO: to verify whether we need the password here, check how multi-srp is handled in the keyring first.
+  ): Promise<void> {
+    // verify the password and unlock the vault
+    const { toprfEncryptionKey, toprfAuthKeyPair } =
+      await this.#verifyPassword(password);
+
+    // encrypt and store the seed phrase backup
+    await this.#encryptAndStoreSeedPhraseBackup(
+      seedPhrase,
+      toprfEncryptionKey,
+      toprfAuthKeyPair,
+    );
+  }
+
+  /**
+   * Add array of new seed phrase backups to the metadata store in batch.
+   *
+   * @param seedPhrases - The seed phrases to backup.
+   * @param password - The password used to create new wallet and seedphrase
+   * @returns A promise that resolves to the success of the operation.
+   */
+  async batchAddSeedPhraseBackups(seedPhrases: Uint8Array[], password: string) {
```

Comment:
i think we should move this function is separate PR to reduce this size of PR, as it is not used in phase 1.